### PR TITLE
use public subnets for rack unless balancer is internal

### PR DESCRIPTION
**Pull request by ddollar**

### CODE REVIEW

File: api/models/templates/app.tmpl [hunk 1]
```diff
@@ -382,18 +382,16 @@
...
-          { "Ref": "Subnets" }
-        ] },
+        {{ if eq .Scheme "internal" }}
+          "Scheme": "internal",
+          "Subnets": { "Fn::If": [ "Private",
+            { "Ref": "SubnetsPrivate" },
+            { "Ref": "Subnets" }
+          ] },
+        {{ else }}
+          "Scheme": { "Fn::If": [ "Internal", "internal", { "Ref": "AWS::NoValue" } ] },
```
Comment (beedub) [inline api/models/templates/app.tmpl]: isn't this unnecessary given that scheme == internal is checked above?

Comment (ddollar) [inline api/models/templates/app.tmpl]: There is the CF parameter "Internal" and the local check of .Scheme == "internal" (the latter is set when you use only a single port pair)

File: api/models/templates/app.tmpl [hunk 2]
```diff
@@ -382,18 +382,16 @@
...
+          "Subnets": { "Fn::If": [ "Internal", { "Ref": "SubnetsPrivate" }, { "Ref": "Subnets" } ] },
```
Comment (beedub) [inline api/models/templates/app.tmpl]: same vein as above, couldn't this be simplified to `"Subnets": { "Ref": "Subnets" }`?

File: api/models/templates/app.tmpl (hunk 1 above)
Comment (beedub) [inline api/models/templates/app.tmpl]: ah gotcha

**Pull request closed without merging**